### PR TITLE
Synchronize app command parser with mobile app

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR aims to synchronize the mobile app's command parser with the webapp, by using common `app_command_parser.ts` and `app_command_parser.test.ts` files. Each platform implements a `app_command_parser_dependencies.ts` file and a `app_command_parser_test_dependencies.ts` file.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-33717

### CODE REVIEW

**Review by levb — APPROVED** [review]

File: actions/apps.ts
Comment (larkox) [inline actions/apps.ts:20]: Should we use it? This could be an improvement on component bindings since they will use the component intl, which from what I understand, is more reliable. And for commands or other situations, we can keep using the intlShim.

Finally, it will allow us to use everywhere the formatMessage, which might be more comprehensible and better tailored for the i18n extract.

**Review by larkox — APPROVED** [review]
There are a few potential changes. In general looks good to me, but I still foresee a few merge conflicts :sweat_smile:

File: components/suggestion/command_provider/app_command_parser/tests/app_command_parser_test_data.ts
Comment (mickmister) [reply components/suggestion/command_provider/app_command_parser/tests/app_command_parser_test_data.ts:97]: @larkox I didn't realize that they are different between mobile and webapp, I may have made a mistake. How are they different?

Comment (larkox) [reply components/suggestion/command_provider/app_command_parser/tests/app_command_parser_test_data.ts:97]: Oh sorry, I meant the other way around.
I thought the objective of `app_command_parser_test_data.ts` was to keep platform dependent code, the same way as `app_command_parser_dependencies.ts`. If that is not the purpose of this file, feel free to ignore my comment.

File: actions/apps.ts
Comment (mickmister) [reply actions/apps.ts]: Using it now

File: components/suggestion/command_provider/app_command_parser/app_command_parser_dependencies.ts
Comment (mickmister) [reply components/suggestion/command_provider/app_command_parser/app_command_parser_dependencies.ts]: Done

**Review by larkox — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
I have spoken with Michael who has done some regression testing locally on the command parser and other affected areas. 

Given that the command parser on webapp is already not quite functional and this PR is holding up other work I'm okay with testing this after it's merged.

**Pull request merged**
